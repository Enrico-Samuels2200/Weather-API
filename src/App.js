import React from 'react';
import 'isomorphic-fetch';
import './App.css';
import Form from './Form';

export default class App extends React.Component {
  constructor(props){
    super(props);
    this.state = { //Creates the state for the Class.
      city: "",
      weather_type: "",
      min_temperature: 0 + "째C",
      max_temperature: 0 + "째C",
      wind_speed: 0 + " km/h",
    }
  }

  fetchWeather = async(val) => { //Gets event data from form input.
    val.preventDefault(); //Prevents the form from refreshing on submit.

    const city = val.target.city.value; //Uses the cities name and save a as constant.

    const res = await fetch(`https://api.openweathermap.org/data/2.5/weather?q=${city}&appid=4189d4a420d6e706009863c17d510bfd`); //fetches the API based on the const "city" and API key.
    const get = await res.json(); //Converts the API from a string into code. 
      try {
        this.setState({ //Sets the state to the chosen converted data.
          city: get.name,
          weather_type: get.weather[0].description,
          max_temperature: (get.main.temp_max - 273.15).toFixed(0) + "째C", //Converts the from kelvin to degrees celsius and put it no decimal places.
          min_temperature: (get.main.temp_min - 273.15).toFixed(0) + "째C",
          wind_speed: (get.wind.speed * 1.852).toFixed(0) + " km/h", //Places wind speed to no decimal places.
        })
      }
      catch(err){ //If there is an error retrieving the data or it's not available it will return this.
        this.setState({ //Sets the state.
          city: "Doesn't exist", //Displays if there is an error in recieving the data or if data doesn't exist.
          weather_type: "",
          min_temperature: 0,
          max_temperature: 0,
          wind_speed: 0,
        })
      }
    }

  render(){
      return (
        <div>
          <h1>{this.state.city}</h1> {/*Retrieves name from the state.*/}
          <Form loadWeather={this.fetchWeather}/> {/*Sets loadWeather value to that of the inputted value of fetchWeather.*/}
          <h1 id="W_P">Weather Type</h1>
          <h2>{this.state.weather_type}</h2>{/*Retrieves weather from the state.*/}
          <hr/>
          <h1 id="Max_T">Max Temperature</h1>
          <h2>{this.state.max_temperature}</h2>
          <hr/>
          <h1 id="Min_T">Min Temperature</h1>
          <h2>{this.state.min_temperature}</h2>
          <hr/>
          <h1 id="W_S">Wind Speed</h1>
          <h2>{this.state.wind_speed}</h2>
        </div>
      );
    
    }
  }