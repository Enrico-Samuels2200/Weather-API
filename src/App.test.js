import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

test.todo('It executed.'), got => {
  function callBack(get){
    expect(data).toBe()
    done();
  }
  callBack()
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div)
};

test("It successfully got the data.", () => {
  const component = renderer.create()
  let app = component.toJSON;
  expect(app).toMatchSnapshot();
})