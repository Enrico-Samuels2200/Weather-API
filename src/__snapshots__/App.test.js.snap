// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First Snapshot test. 1`] = `
<div>
  <h1>
    
  </h1>
   
  <div>
    <form
      className="inputForm"
      onSubmit={[Function]}
    >
      <input
        autoComplete="off"
        name="city"
        placeholder="Enter city name"
        type="text"
      />
       
      <button>
        Get Weather
      </button>
      <hr />
    </form>
  </div>
   
  <h1
    id="W_P"
  >
    Weather Type
  </h1>
  <h2>
    
  </h2>
  <hr />
  <h1
    id="Max_T"
  >
    Max Temperature
  </h1>
  <h2>
    0°C
  </h2>
  <hr />
  <h1
    id="Min_T"
  >
    Min Temperature
  </h1>
  <h2>
    0°C
  </h2>
  <hr />
  <h1
    id="W_S"
  >
    Wind Speed
  </h1>
  <h2>
    0 km/h
  </h2>
</div>
`;

exports[`It successfully got the data. 1`] = `[Function]`;
